components/org.wso2.carbon.identity.inbound.ldap/src/main/java/org/wso2/carbon/identity/inbound/ldap/internal/LDAPEPComponent.java

```diff
@@ -0,0 +1,197 @@
+import org.wso2.carbon.user.core.service.RealmService;
+
+import java.io.File;
+
+
+@Component(
+        name = "org.wso2.carbon.identity.inbound.ldap.internal.LDAPEPComponent",
+        immediate = true)
+public class LDAPEPComponent {
+
+    private static final Log log = LogFactory.getLog(LDAPEPComponent.class);
+
+    private LDAPServer ldapServer;
+
+    @Activate
+    public void activate(ComponentContext context) {
+
+        try {
+
+            /*Read the ldap-org.wso2.carbon.identity.inbound.ldap.endpoint configuration file.*/
+
+            ConfigurationBuilder configurationBuilder = new ConfigurationBuilder();
+
+            configurationBuilder.buildConfiguration(getLdapConfigurationFile());
+
+            /*Make relevant objects that encapsulate different parts of config file.*/
+            configurationBuilder.buildLDAPEndpointConfigurartion();
+
+            boolean LdapEndpointEnabled = configurationBuilder.isLdapEndpointEnabled();
+            //expose LDAP Endpoint only if ldap-org.wso2.carbon.identity.inbound.ldap.endpoint is enabled.
+            if (LdapEndpointEnabled) {
+
+                EndPointConfiguration endPointConfiguration = configurationBuilder.getEndPointConfiguration();
+
+                /*set the embedded-apacheds's schema location which is: carbon-home/repository/data/
+                  is-default-schema.zip
+                */
+                setSchemaLocation();
+
+                /* Set working directory where schema directory and ldap partitions are created*/
+                setWorkingDirectory(endPointConfiguration);
+
+                startLdapServer(endPointConfiguration);
+
+                /* replace default password with that is provided in the configuration file.*/
+                this.ldapServer.changeConnectionUserPassword(configurationBuilder.getConnectionPassword());
+
+
+                if (log.isDebugEnabled()) {
+                    log.debug("apacheds-enpoint component started.");
+                }
+
+            } else if (!LdapEndpointEnabled) {
+                //if needed, create a dummy tenant manager service and register it.
+                log.info("LDAP Endpoint is disabled.");
+            }
+
+        } catch (Exception e) {
+            log.error("Could not start the ldap-org.wso2.carbon.identity.inbound.ldap.endpoint. ", e);
+        }
+
+    }
+
+    @Deactivate
+    public void deactivate(ComponentContext context)
+            throws Exception {
+        if (this.ldapServer != null) {
+
+            this.ldapServer.stop();
+        }
+    }
+
+    private void setSchemaLocation() throws Exception {
+        String schemaLocation = "repository" + File.separator + "data" + File.separator +
+                "is-default-schema.zip";
+        File dataDir = new File(getCarbonHome(), schemaLocation);
+
+        // Set schema location
+        System.setProperty("schema.zip.store.location", dataDir.getAbsolutePath());
```

Comment:
Make 'schema.zip.store.location' a constant and make it fully-qualified to avoid any conflicts. You can simply prefix this with 'apacheds' text.

Reply:
Fixed this one.